### DISCUSSION TITLE
DeDRM error: Plugin is invalid.

### DISCUSSION THREAD

**Discussion by scubacat22** [noDRM/DeDRM_tools#220] [General]
Morning! I had to upgrade Calibre to 6.9 because it wasn't detecting my new Kindle 11th generation. So I found the newest version of DeDRM (version 10.0.3) but when I try and "load plugin from file" I get an error that says the plugin is invalid. It does not contain a top-level _init_.py file. Any suggestions? I did search for this issue, found a comment that said you needed to download the files from a particular site, so I went there and tried again, but got the same result. 

Are my versions of Calibre and DeDRM incompatible? Does anyone know the magic formula for kindle recognition AND plugin? Thanks!

**Comment by ElleKayEm** [comment]
Sounds like you're trying to load the wrong file.  Go here:  https://github.com/noDRM/DeDRM_tools/releases/tag/v10.0.3

Download DeDRM_tools_10.0.3.zip (1st link under Assets).  Unzip it and load DeDRM_plugin.zip.

**Reply by scubacat22** [reply]
That's where I got it from. Do I need to unzip it into a particular place? It's currently just hanging out in my downloads folder. 
When I tell Calibre to open from file and select DeDRM_tools_10.0.3.zip, I get a security risk warning, I click Yes. I get the invalidplugin error. 
When I tell Calibre to open from file and open the DeDRM_tools_10.0.3 folder, navigate to DeDRM_plugin.zip, and select that, nothing happens. No warning about security. No error. But nothing else happens. 
When I go to the downloaded DeDRM_tools_10.0.3.zip, click extract all files, I then tell Calibre to open from file, and navigate to the DeDRM_plugin_zip file, I get a security warning. Click yes, and then I get a ModuleNotFoundError: No module named '_version'

That's at least a new error! ;)

**Reply by scubacat22** [reply]
I may have actually got it working. I went through and deleted all my previous downloads of these files, and started fresh and it might have worked this time! I will let you know. :) Thanks

**Reply by kstingel** [reply]
I had the same error
Solution: 
unzip the container archive to it's own folder, 
there are 2 plugin archives inside it - install each of these into Calibre
